core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3094,14 +3094,14 @@ protected Scope visitUpdate(Update update, Optional<Scope> scope)
                     .collect(toImmutableMap(ColumnSchema::getName, Function.identity()));
 
             for (UpdateAssignment assignment : update.getAssignments()) {
-                String columnName = assignment.getName().getValue();
+                String columnName = assignment.getName().getValue().toLowerCase(ENGLISH);
```

Comment:
This needs to respect the semantics of delimited (quoted) identifiers.

Reply:
Thanks for pointing this out but does Trino actually respect this semantics? I found https://github.com/trinodb/trino/issues/17 which seems still to do. Additionally, I've added a test and quoted identifiers seem to make no difference except for `ALTER TABLE table EXECUTE prodedure_name` where quoting of `procedure_name` is actually respected.